Network.hs

```diff
@@ -292,6 +293,7 @@ accept sock@(MkSocket _ AF_INET6 _ _ _) = do
 # if !defined(mingw32_HOST_OS)
                  SockAddrUnix      a   -> return a
 # endif
+                 _ -> error "accept: socket address not supported on this platform."
```

Comment:
So in this case things are a bit tricky: the catchall won't catch just the `SockAddrCan` case, which isn't supported in `accept`, but also `SockAddrUnix`, but only on Windows. So if I remove the wildcard we'll be getting many duplicate patterns on the LHS, and duplication of the RHS too, but I'll give it a try.